### PR TITLE
RHDEVDOCS-2098: Integrated definitions with concepts

### PR DESCRIPTION
**Pull request by HarshCasper**
JIRA ID: [RHDEVDOCS-2098](https://issues.redhat.com/browse/RHDEVDOCS-2098)

Direct Preview Link: https://deploy-preview-31606--osdocs.netlify.app/openshift-enterprise/latest/cicd/pipelines/understanding-openshift-pipelines.html
QE review done
Aligned team: Dev Tools

### CODE REVIEW

**Comment by Preeticp** [comment]
@HarshCasper please update the description of the PR with the SME/QE review status and direct preview link

File: modules/op-about-taskrun.adoc [hunk 1]
```diff
@@ -5,7 +5,7 @@
 [id="about-taskrun_{context}"]
 = TaskRun
 
-A _TaskRun_ instantiates a Task for execution with specific inputs, outputs, and execution parameters on a cluster. It can be invoked on its own or as part of a PipelineRun.
+A _TaskRun_ instantiates a Task for execution with specific inputs, outputs, and execution parameters on a cluster. It can be invoked on its own or as part of a PipelineRun for each Task in a pipeline.
```
Comment (Preeticp) [inline modules/op-about-taskrun.adoc:8]: Please add this too: A TaskRun is automatically created by a PipelineRun for each Task in a Pipeline.

File: modules/op-about-pipelines.adoc [hunk 2]
```diff
@@ -5,9 +5,9 @@
...
 
-A `Pipeline` resource definition consists of a number of fields or attributes, which together enable the pipeline to accomplish a specific goal. Each `Pipeline` resource definition must contain at least one `Task` resource, which ingests specific inputs and produces specific outputs. The pipeline definition can also optionally include _Conditions_, _Workspaces_, _Parameters_, or _Resources_ depending on the application requirements.
+A `Pipeline` resource definition consists of a number of fields or attributes, which together enable the pipeline to accomplish a specific goal. Each `Pipeline` resource definition must contain at least one `Task` resource, which ingests specific inputs and produces specific outputs. A Pipeline also interacts with the outside world using PipelineResources. The pipeline definition can also optionally include _Conditions_, _Workspaces_, _Parameters_, or _Resources_ depending on the application requirements.
```
Comment (Preeticp) [inline modules/op-about-pipelines.adoc:10]: Drop 'also' - A Pipeline interacts with the outside world using PipelineResources.

**Review by Preeticp — COMMENTED** [review]
@HarshCasper a couple of comments and this is good to be merged.

File: modules/op-about-pipelines.adoc (hunk 2 above)
Comment (HarshCasper) [reply modules/op-about-pipelines.adoc:10]: Done!

File: modules/op-about-taskrun.adoc (hunk 1 above)
Comment (HarshCasper) [reply modules/op-about-taskrun.adoc:8]: Done!

File: modules/op-about-pipelines.adoc [hunk 3]
```diff
@@ -5,9 +5,9 @@
...
 
-A `Pipeline` resource definition consists of a number of fields or attributes, which together enable the pipeline to accomplish a specific goal. Each `Pipeline` resource definition must contain at least one `Task` resource, which ingests specific inputs and produces specific outputs. The pipeline definition can also optionally include _Conditions_, _Workspaces_, _Parameters_, or _Resources_ depending on the application requirements.
+A `Pipeline` resource definition consists of a number of fields or attributes, which together enable the pipeline to accomplish a specific goal. Each `Pipeline` resource definition must contain at least one `Task` resource, which ingests specific inputs and produces specific outputs. A Pipeline interacts with the outside world using PipelineResources. The pipeline definition can also optionally include _Conditions_, _Workspaces_, _Parameters_, or _Resources_ depending on the application requirements.
```
Comment (ppitonak) [inline modules/op-about-pipelines.adoc:10]: PipelineResources are deprecated and we removed them in multiple places in 1.4. What was the motivation to add this sentence?

File: modules/op-about-tasks.adoc [hunk 4]
```diff
@@ -5,9 +5,9 @@
 [id="about-tasks_{context}"]
 = Tasks
 
-_Tasks_ are the building blocks of a Pipeline and consist of sequentially executed Steps. Tasks are reusable and can be used in multiple Pipelines.
+_Tasks_ are the building blocks of a Pipeline and consists of sequentially executed steps. It is essentially a function of inputs and outputs that form the Pipeline build. A Task can run individually or as a part of the pipeline. Tasks are reusable and can be used in multiple Pipelines.
```
Comment (ppitonak) [inline modules/op-about-tasks.adoc:8]: 1. Tasks and pipelines are sometimes capitalized, sometimes not, shouldn't we unify that?
2. `that form the Pipeline build`... I'm not sure what you want to say here

**Review by ppitonak — CHANGES_REQUESTED** [review]

File: modules/op-about-tasks.adoc (hunk 4 above)
Comment (Preeticp) [reply modules/op-about-tasks.adoc:8]: We will be creating another PR to tackle all the capitalization of the CRD, post the release rush. We will make all of them small case as per the OCP docs guidance.

File: modules/op-about-pipelines.adoc (hunk 3 above)
Comment (Preeticp) [reply modules/op-about-pipelines.adoc:10]: Thanks Pavol. This was part of the definitions section, we will remove it.

File: modules/op-about-tasks.adoc (hunk 4 above)
Comment (Preeticp) [reply modules/op-about-tasks.adoc:8]: @HarshCasper we can remove this sentence: It is essentially a function of inputs and outputs that form the Pipeline build.
The task run part explains this in a better manner.

File: cicd/pipelines/understanding-openshift-pipelines.adoc [hunk 5]
```diff
@@ -23,9 +22,6 @@ include::modules/technology-preview.adoc[leveloffset=+1]
 * You can use {pipelines-title} to build images with Kubernetes tools such as Source-to-Image (S2I), Buildah, Buildpacks, and Kaniko that are portable across any Kubernetes platform.
 * You can use the {product-title} Developer Console to create Tekton resources, view logs of Pipeline runs, and manage pipelines in your {product-title} namespaces.
 
-//Main Concept Topic
-include::modules/op-pipelines-concepts.adoc[leveloffset=+1]
-
 [id="op-detailed-concepts"]
 == Detailed OpenShift Pipeline Concepts
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply cicd/pipelines/understanding-openshift-pipelines.adoc:26]: Done!

File: modules/op-about-pipelines.adoc (hunk 3 above)
Comment (HarshCasper) [reply modules/op-about-pipelines.adoc:10]: Done!

File: modules/op-about-tasks.adoc (hunk 4 above)
Comment (HarshCasper) [reply modules/op-about-tasks.adoc]: Done!

**Pull request merged**